src/Components/Pages/SetNamePage.js

```diff
@@ -10,10 +10,16 @@ import { HelperText } from '@patternfly/react-core/dist/dynamic/components/Helpe
 import { HelperTextItem } from '@patternfly/react-core/dist/dynamic/components/HelperText';
 import { Form } from '@patternfly/react-core/dist/dynamic/components/Form';
 
-const SetNamePage = ({ name, setName, nameIsValid }) => {
+const SetNamePage = ({ mode, name, setName, nameIsValid, isNameDisabled }) => {
   const [enableValidationFeedback, setEnableValidationFeedback] =
     useState(false);
 
+  useEffect(() => {
+    if (mode === 'edit' && name) {
+      setName(name);
+    }
+  }, [mode, name, setName]);
+
```

Comment:
Polish: Remove this `useEffect` - it doesn't do anything. `name` is managed state by another component and so the only way it has ever changed is if `setName` has already been called

Reply:
If it still works, we remove dependence on knowing the context of the modal from this page

Reply:
Actually looking some more, it looks like `mode` is not passed to this component in `SetNameAndDescriptionPage` already, so I'm even more confident that it is unneeded in this component